services/bigpanda/service.go

```diff
@@ -215,24 +218,45 @@ func (s *Service) preparePost(appKey, id string, message string, details string,
 	bpData["timestamp"] = timestamp.Unix()
 	bpData["status"] = status
 
-	if appKey == "" {
-		appKey = c.AppKey
+	// primary and secondary property
+	if hc.PrimaryProperty != "" {
+		bpData["primary_property"] = hc.PrimaryProperty
+	}
+	if hc.SecondaryProperty != "" {
+		bpData["secondary_property"] = hc.SecondaryProperty
+	}
+
+	if hc.AppKey != "" {
+		bpData["app_key"] = hc.AppKey
+	} else {
+		bpData["app_key"] = c.AppKey
 	}
-	bpData["app_key"] = appKey
 
 	if len(data.Tags) > 0 {
 		for k, v := range data.Tags {
 			bpData[k] = v
 		}
 	}
 
+	if len(data.Fields) > 0 {
+		for k, v := range data.Fields {
+			bpData[k] = fmt.Sprintf("%v", v)
+		}
+	}
+
 	var post bytes.Buffer
 	enc := json.NewEncoder(&post)
 	if err := enc.Encode(bpData); err != nil {
 		return nil, err
 	}
 
-	alertUrl, err := url.Parse(c.URL)
+	var bpUrl string
+	if hc.URL != "" {
+		bpUrl = hc.URL
+	} else {
+		bpUrl = c.URL
+	}
```

Comment:
Since strings in go are immutable, we can do this which is a bit cleaner.

```go
    bpUrl := hc.URL
    if bpUrl == "" {
        bpUrl = c.URL
}
```